Part_2_/Person.cs

```diff
@@ -1,11 +1,68 @@
-﻿namespace Part_2
+﻿
+namespace Part_2
 {
     // TODO: inherit IPerson interface
     // TODO: create Person class with public parameters Id (long), FirstName (string), LastName (string), Email (string), Gender (Enum).
     // TODO: add method to get full name()
     // TODO: validate an email, email validation: check for NULL or Empty and contains string = "@". OnFail - thrown an ArgumentException.
     // TODO: override Equals and  GetHashCode
-    public class Person
+
+    public class Person:IPerson
     {
+        private string email;
+        public long Id { get; set; }
+        public string FirstName { get; set; }
+        public string LastName { get; set; }
+        public string Email {
+            get=>email;
+            set
+            {
+                if (string.IsNullOrEmpty(value) | !value.Contains("@")) throw new System.ArgumentException();
+                email = value;
+            }
+        }
+        public Gender Gender { get; set; }
+
+        public int CompareTo(IPerson other)
+        {
+            if (other == null) return 1;
+            else 
+            {
+                int returnCode = 0;// не знал как лучше назвать
+                returnCode = this.Id.CompareTo(other.Id);
+                if (returnCode != 0) return returnCode;
+                returnCode = this.GetFullName().CompareTo(other.GetFullName());
+                if (returnCode != 0) return returnCode;
+                returnCode = this.Email.CompareTo(other.Email);
+                if (returnCode != 0) return returnCode;
+                returnCode = this.Gender.CompareTo(other.Gender);
+                return returnCode;
+            }
+
+        }
+
+        public int CompareTo(object obj) => obj == null ? 1 : CompareTo(obj as IPerson);
+
+        public string GetFullName() => FirstName + LastName;
+        public override bool Equals(object obj)
+        {
+            if (!(obj is IPerson person)) return false;
```

Comment:
please use logical operators
and all this conditions returns boolean, you don't need to check and return
and one more point it is better to use success conditions (means check for true if it possible), can be checked for false to not using huge blocks in if statements
you can use like this:

return obj is IPerson && Id == person.Id && ...